src/core/coap/coap.cpp

```diff
@@ -838,7 +800,7 @@ void ResponsesQueue::EnqueueResponse(const Header &aHeader, Message &aMessage, c
         DequeueOldestResponse();
     }
 
-    copy = aMessage.Clone();
+    copy = static_cast<Message *>(aMessage.Clone());
```

Comment:
Is this `static_cast` necessary with the introduction of `ot::Coap::Message::Clone()` methods?